include/pm/device_runtime.h

```diff
@@ -130,13 +137,33 @@ int pm_device_runtime_put_async(const struct device *dev);
  */
 bool pm_device_runtime_is_enabled(const struct device *dev);
 
+/**
+ * @brief Initialize a device state to #PM_DEVICE_STATE_SUSPENDED.
+ *
+ * By default device state is initialized to #PM_DEVICE_STATE_ACTIVE. In
+ * general, this makes sense because the device initialization function will
+ * resume and configure a device, leaving it operational. However, when device
+ * runtime PM is enabled, the device may not be resumed and the init function
+ * will just enable device runtime PM. If that is the case, this function can be
+ * used to set the initial device state to #PM_DEVICE_STATE_SUSPENDED.
+ *
+ * @param dev Device instance.
+ */
+static inline void pm_device_runtime_init_suspended(const struct device *dev)
+{
+	struct pm_device *pm = dev->pm;
+
+	pm->state = PM_DEVICE_STATE_SUSPENDED;
+}
```

Comment:
I don't think this function is right, we are trying to fix an inconsistency in the wrong place. My point is, if a device is physically suspended but its state is saying it is ACTIVE, it not only a runtime pm issue, because if you are not using pm runtime the system when going to sleep will look this device state and call `pm->action_cb`(pm->dev, PM_DEVICE_ACTION_SUSPEND);` for it, what is the same thing we are trying to avoid here.

Am I wrong ?

Reply:
The thing is that all devices are initialized as `PM_DEVICE_STATE_ACTIVE` (assuming init hook will leave it in this state). However, if we consider the STM32 GPIO case where the device is never clocked, we need such mechanism so that `pm_device_runtime_enable` doesn't try to suspend it.

Reply:
This smells workaround in the wrong place. What happens with this device when runtime pm is disabled (built without it) ? It initializes not clocked but the state is active, then when the system gets idle, this device is suspended and when the system wakes it is resumed. In this scenario you 
will have  pm->action_cb(pm->dev, PM_DEVICE_ACTION_SUSPEND) called with the device not clocked.

Anyway, I think we can live with this for now, since the problem is there and this solves at least for this use case. I just don't think this is the proper solution for the problem.

Reply:
nevermind, I saw that in case it is built without device runtime pm you the device is clocked ...